Add descriptions for every project

nkolev92 opened the pull request:
## Bug

Fixes: https://github.com/NuGet/Client.Engineering/issues/205
Regression: No  
* Last working version:   
* How are we preventing it in future:   

## Fix

Details: 
In order to generate better overview of our code structure we need better project descriptions.

## Testing/Validation

Tests Added: No  
Reason for not adding tests: No functional changes
Validation: Automation that it builds :)

donnie-msft commented on src/NuGet.Clients/NuGet.Console/NuGet.Console.csproj:
```diff
@@ -10,13 +10,9 @@
-    <PackagesDirectory>$(UserProfile)\.nuget\packages</PackagesDirectory>
-    <ResolveNuGetPackages>true</ResolveNuGetPackages>
-    <SkipValidatePackageReferences>true</SkipValidatePackageReferences>
+    <Description>Package Manager Powershell Console implementation</Description>
```
typo: PowerShell

donnie-msft commented on src/NuGet.Clients/NuGet.Console/Properties/AssemblyInfo.cs:
```diff
@@ -10,7 +10,7 @@
 
 [assembly: AssemblyTitle("NuGet.Console")]
-[assembly: AssemblyDescription("Package Manager Powershell Console")]
+[assembly: AssemblyDescription("Package Manager Powershell Console implementation")]
```
typo: PowerShell

donnie-msft commented on src/NuGet.Clients/NuGet.VisualStudio.Common/NuGet.VisualStudio.Common.csproj:
```diff
@@ -10,6 +10,7 @@
     <GenerateDocumentationFile>false</GenerateDocumentationFile>
     <RootNamespace>NuGet.VisualStudio</RootNamespace>
     <NoWarn>$(NoWarn);NU5104</NoWarn>
+    <Description>NuGet's Visual Studio common types and interfaces used for both Package Manager and Package Manager Console</Description>
```
Maybe add "UI" after the first Manager?

nkolev92 replied:
Add UI and some additional explanations.

donnie-msft commented on src/NuGet.Core/NuGet.Build.Tasks.Pack/NuGet.Build.Tasks.Pack.csproj:
```diff
@@ -17,7 +17,7 @@
     <SuppressDependenciesWhenPacking>true</SuppressDependenciesWhenPacking>
     <DevelopmentDependency>true</DevelopmentDependency>
-    <PackageDescription>NuGet pack for dotnet CLI.</PackageDescription>
+    <Description>NuGet Build tasks for MSBuild and dotnet pack</Description>
```
Add ticks around dotnet pack like you did for `dotnet nuget` ?

nkolev92 replied:
It's MSBuild and dotnet pack, sot he pack refers to both. I've actually decided to go the other way. 
Removed the ticks altogether, let me know what you think.

donnie-msft commented on src/NuGet.Core/NuGet.Build.Tasks/NuGet.Build.Tasks.csproj:
```diff
@@ -8,7 +8,7 @@
     <IncludeInVSIX>true</IncludeInVSIX>
     <PackProject>true</PackProject>
-    <Description>NuGet restore for dotnet CLI.</Description>    
+    <Description>NuGet Build tasks for MSBuild and dotnet restore</Description>    
```
Add ticks ` around dotnet restore like you did for `dotnet nuget` ?

nkolev92 replied:
Similar to above, the restore part refers to both MSBuild and dotnet.

zivkan replied:
I think customers will find this a bit confusing, as it's not really about "build", it's about "restore". Ok, in .NET Core 2.x and above there's an implicit restore as part of build, but I think calling it "restore tasks", rather than "build tasks" would be clearer.

Also, I'm not sure why build has a capital B.

donnie-msft commented on src/NuGet.Core/NuGet.Packaging.Core/NuGet.Packaging.Core.csproj:
```diff
@@ -3,7 +3,7 @@
 
   <PropertyGroup>
-    <Description>The (former home to) core data structures for NuGet.Packaging.</Description>
+    <Description>The (former home to) core data structures for NuGet.Packaging</Description>
```
Why do we still have this assembly? Can the description explain why we have it instead of just saying it's the former?

nkolev92 replied:
Added an extra line.

donnie-msft commented on src/NuGet.Core/NuGet.ProjectModel/NuGet.ProjectModel.csproj:
```diff
@@ -10,6 +10,7 @@
     <Shipping>true</Shipping>
     <IncludeInVSIX>true</IncludeInVSIX>
     <XPLATProject>true</XPLATProject>
+    <Description>NuGet's core types and interfaces for PackageReference based restore, such as lock files, assets file and internal restore models</Description>
```
typo (add hyphen): PackageReference-based

donnie-msft commented on test/NuGet.Clients.Tests/NuGetConsole.Host.PowerShell.Test/NuGetConsole.Host.PowerShell.Test.csproj:
```diff
@@ -6,6 +6,7 @@
     <TargetFramework>$(NETFXTargetFramework)</TargetFramework>
     <TestProject>true</TestProject>
     <UseParallelXunit>true</UseParallelXunit>
+    <Description>Unit and integration tests for NuGetConsole.Host.Powershell</Description>
```
typo: PowerShell

donnie-msft commented on test/NuGet.Core.FuncTests/Dotnet.Integration.Test/Dotnet.Integration.Test.csproj:
```diff
@@ -5,6 +5,7 @@
   <PropertyGroup>
     <TargetFrameworks>$(NETCoreTargetFramework)</TargetFrameworks>
     <TestProject>true</TestProject>
+    <Description>Integration tests for NuGet powered dotnet.exe comands such as pack/restore/list package and dotnet nuget</Description>
```
typo (add hyphen): NuGet-powered
Also maybe add tick marks around `dotnet nuget` like other places.

nkolev92 replied:
I only have one `dotnet nuget` and I have removed it.

I don't think I understand why I need the hyphen here in NuGet-powered?

donnie-msft replied:
You did not remove dotnet nuget at the end of the sentence.
I think "powered" on its own doesn't make sense. I think you're making it a compound word.

nkolev92 replied:
> You did not remove dotnet nuget at the end of the sentence.

Why do I need to remove it? 

Added a hyphen for NuGet-powered

donnie-msft commented on test/NuGet.Core.FuncTests/NuGet.Commands.FuncTest/NuGet.Commands.FuncTest.csproj:
```diff
@@ -7,6 +7,7 @@
     <TargetFrameworks Condition=" '$(IsXPlat)' == 'true' ">$(NETCoreTargetFramework)</TargetFrameworks>
     <TestProject>true</TestProject>
     <TestProjectType>functional</TestProjectType>
+    <Description>Integration tests for the more involved NuGet.Commands like restore</Description>
```
typo: ", such as restore" (comma plus consistency with other verbiage)

donnie-msft commented on test/NuGet.Core.FuncTests/NuGet.Packaging.FuncTest/NuGet.Packaging.FuncTest.csproj:
```diff
@@ -7,6 +7,7 @@
     <TargetFrameworks Condition=" '$(IsXPlat)' == 'true' ">$(NETCoreTargetFramework)</TargetFrameworks>
     <TestProject>true</TestProject>
     <UseParallelXunit>true</UseParallelXunit>
+    <Description>Integration tests for the more involved NuGet.Packaging functionality such as signing</Description>
```
typo: add a comma before "such as"

donnie-msft commented on test/NuGet.Core.Tests/Microsoft.Build.NuGetSdkResolver.Tests/Microsoft.Build.NuGetSdkResolver.Test.csproj:
```diff
@@ -7,6 +7,7 @@
     <TargetFrameworks Condition=" '$(IsXPlat)' == 'true' ">$(NETCoreTargetFramework)</TargetFrameworks>
     <TestProject>true</TestProject>
     <UseParallelXunit>true</UseParallelXunit>
+    <Description>Unit for Microsoft.Build.NuGetSdkResolver</Description>
```
typo: Unit tests

donnie-msft commented on test/NuGet.Tests.Apex/NuGet.Console.TestContract/NuGet.Console.TestContract.csproj:
```diff
@@ -6,6 +6,7 @@
     <TargetFramework>$(NETFXTargetFramework)</TargetFramework>
     <OutputType>Library</OutputType>
     <NoWarn>$(NoWarn);CS1998</NoWarn>
+    <Description>A contract assembly providing a wrapper for testing the NuGet Powershell functionality</Description>
```
typo: PowerShell

donnie-msft commented on test/NuGet.Tests.Apex/NuGet.Tests.Apex/NuGet.Tests.Apex.csproj:
```diff
@@ -7,6 +7,7 @@
     <NETCoreWPFProject>true</NETCoreWPFProject>
     <TestProject>true</TestProject>
     <NoWarn>$(NoWarn);1762</NoWarn>
+    <Description>End to end tests for the Package Manager Console and Package Manager UI built on top of apex framework</Description>
```
typo: hyphenate end-to-end
typo: Apex capitalized?

donnie-msft commented on test/TestExtensions/API.Test/API.Test.csproj:
```diff
@@ -5,6 +5,7 @@
     <TargetFramework>$(NETFXTargetFramework)</TargetFramework>
     <SkipSigning>true</SkipSigning>
     <NoWarn>$(NoWarn);1762</NoWarn>
+    <Description>An integration wrapper over the NuGet extensibility APIs for the PMC based End to End tests</Description>
```
typo: hyphenate end-to-end
PMC should be spelled out.

donnie-msft commented on test/TestExtensions/GenerateTestPackages/GenerateTestPackages.csproj:
```diff
@@ -8,6 +8,7 @@
     <SignAssembly>true</SignAssembly>
     <AssemblyOriginatorKeyFile>ToolTestKey.snk</AssemblyOriginatorKeyFile>
     <DelaySign>false</DelaySign>
+    <Description>A utility for generating test packages for End to End and Apex tests</Description>
```
typo: hyphenate end-to-end

donnie-msft commented on test/TestExtensions/SampleCommandLineExtensions/SampleCommandLineExtensions.csproj:
```diff
@@ -4,9 +4,8 @@
-  </PropertyGroup>
-  <PropertyGroup>
     <RootNamespace>SampleCommandLineExtensions</RootNamespace>
+    <Description>A sample commandline extensions for testing NuGet.exe command extensibility</Description>
```
typo: A sample **of** **command-line**

nkolev92 replied:
Don't think that reads super well. I went with
`A sample commandline extension for testing NuGet.exe command extensibility` 

Let me know what you think.

AraHaan replied:
I think it would be better if it got changed to ``A sample command-line extension for testing NuGet.exe command extensibility.``

donnie-msft replied:
Still prefer the hypen as well.

nkolev92 replied:
ok, fixed.

donnie-msft commented on test/TestUtilities/Test.Utility/Test.Utility.csproj:
```diff
@@ -11,6 +11,7 @@
     <IsPackable>true</IsPackable>
     <SkipShared>true</SkipShared>
     <NoWarn>$(NoWarn);VSTHRD001;VSTHRD002;VSTHRD110</NoWarn>
+    <Description>A collection of test utilities liek generating packages, mocking servers, stub implementations of interfaces, etc</Description>
```
typo: like (but maybe would prefer "such as" to match your other verbiage)

donnie-msft reviewed (changes requested):
👏 So awesome to see summaries of these assemblies starting to come together!
I also see you took the opportunity to cleanup some (I assume) unused properties.

I've tried to provide touch-ups for consistency.
For the Unit Tests, I'd like us to eventually have more detailed description as to what we are uniquely expecting from each test assembly (some sound similar which is confusing), but for now it's a good start we can iterate on.

nkolev92 reviewed (commented):
@donnie-msft Addressed your feedback.
Resolved all comments which were pretty clear integration of the suggestion, added a comment and the thread unresolved for the ones that need further discussion.

zivkan commented on src/NuGet.Clients/NuGet.CommandLine/NuGet.CommandLine.csproj:
```diff
@@ -15,7 +15,7 @@
     <TargetFramework>$(NETFXTargetFramework)</TargetFramework>
     <GenerateDocumentationFile>false</GenerateDocumentationFile>
-    <Description>NuGet Command Line Tool.</Description>
+    <Description>NuGet Command Line Tool</Description>
```
```suggestion
    <Description>NuGet Command Line Interface</Description>
```

Our docs refer to nuget.exe as "nuget.exe CLI"

zivkan commented on src/NuGet.Clients/NuGet.SolutionRestoreManager/NuGet.SolutionRestoreManager.csproj:
```diff
@@ -13,6 +13,7 @@
     <RootNamespace>NuGet.SolutionRestoreManager</RootNamespace>
     <AssemblyName>NuGet.SolutionRestoreManager</AssemblyName>
     <TargetFrameworkVersion>$(NETFXTargetFrameworkVersion)</TargetFrameworkVersion>
+    <Description>NuGet Solution Restore Manager</Description>
```
```suggestion
    <Description>NuGet's Visual Studio Solution Restore Manager</Description>
```

zivkan commented on src/NuGet.Core/NuGet.CommandLine.XPlat/NuGet.CommandLine.XPlat.csproj:
```diff
@@ -3,7 +3,7 @@
 
   <PropertyGroup>
-    <Description>NuGet wrapper for dotnet.exe.</Description>
+    <Description>NuGet executable wrapper for the dotnet nuget functionality</Description>
```
```suggestion
    <Description>NuGet executable wrapper for the dotnet CLI nuget functionality</Description>
```

zivkan reviewed (approved):
I disagree with the removal of periods at the end of descriptions. Descriptions are English sentences, not simply a title. A few descriptions have more than one sentence, and in those cases the first sentence has a period, but the second does not. I think it would be better that all descriptions, even single sentence descriptions, ends in a period.

heng-liu commented on src/NuGet.Clients/NuGet.Console/NuGet.Console.csproj:
```diff
@@ -10,13 +10,9 @@
     <ProjectGuid>{50E33DA2-AF14-486D-81B8-BD8409744A38}</ProjectGuid>
     <CreateVsixContainer>False</CreateVsixContainer>
     <DeployExtension>False</DeployExtension>
-    <NuGetPackageImportStamp>751ce4af</NuGetPackageImportStamp>
-    <PackagesDirectory>$(UserProfile)\.nuget\packages</PackagesDirectory>
-    <ResolveNuGetPackages>true</ResolveNuGetPackages>
-    <SkipValidatePackageReferences>true</SkipValidatePackageReferences>
```
May I know why these lines are removed?

nkolev92 replied:
They are just unnecessary. 
Some of these were useful when the features were in preview, but that's not the case anymore.

heng-liu replied:
So it was used to restore when some projects in NuGet solution were packages.config, right?

heng-liu commented on src/NuGet.Clients/NuGet.Console/NuGet.Console.csproj:
```diff
@@ -10,13 +10,9 @@
+    <Description>Package Manager PowerShell Console implementation</Description>
     <TargetFrameworkProfile />
     <TargetFrameworkVersion>$(NETFXTargetFrameworkVersion)</TargetFrameworkVersion>
-    <RestoreProjectStyle>PackageReference</RestoreProjectStyle>
```
And why this line is removed? Thanks.

nkolev92 replied:
This one was moved to common.project.props. 

All the projects in our solution are Package Reference based.

heng-liu replied:
👍 Got it! Thanks for clarification.

nkolev92 commented:
> I disagree with the removal of periods at the end of descriptions. Descriptions are English sentences, not simply a title. A few descriptions have more than one sentence, and in those cases the first sentence has a period, but the second does not. I think it would be better that all descriptions, even single sentence descriptions, ends in a period.

Feedback picked up from this review; 

* I will add back the period at the end of all descriptions. 
* Call things `nuget.exe CLI` and `dotnet CLI` where appropriate.

aortiz-msft reviewed (approved):
:shipit:

nkolev92 commented:
@zivkan Addressed your feedback.
Everything description ends with a period `.` and fixed up the descriptions for NuGet.Build.Tasks and NuGet.Build.Tasks.Pack

nkolev92 commented:
@donnie-msft whenever you can please either take another look or dismiss the stale review.

dominoFire commented on test/NuGet.Core.Tests/NuGet.CommandLine.Xplat.Tests/NuGet.CommandLine.Xplat.Tests.csproj:
```diff
@@ -7,6 +7,7 @@
     <TargetFrameworks Condition=" '$(IsXPlat)' == 'true' ">$(NETCoreTargetFramework)</TargetFrameworks>
     <TestProject>true</TestProject>
     <UseParallelXunit>true</UseParallelXunit>
+    <Description>Unit tests for NuGet.CommandLine.Xplat.</Description>
```
```suggestion
    <Description>Unit tests for NuGet.CommandLine.XPlat.</Description>
```

dominoFire commented on test/NuGet.Core.FuncTests/NuGet.XPlat.FuncTest/NuGet.XPlat.FuncTest.csproj:
```diff
@@ -7,6 +7,7 @@
     <TargetFrameworks Condition=" '$(IsXPlat)' == 'true' ">$(NETCoreTargetFramework)</TargetFrameworks>
     <TestProject>true</TestProject>
     <TestProjectType>functional</TestProjectType>
+    <Description>Functionality for the nuget functionality in dotnet CLI, dotnet nuget scenarios. Testing using the NuGet.CommandLine.xplat assembly.</Description>
```
```suggestion
    <Description>Functional tests for nuget in dotnet CLI scenarios, using the NuGet.CommandLine.XPlat assembly.</Description>
```

dominoFire commented on src/NuGet.Clients/NuGet.PackageManagement.PowerShellCmdlets/NuGet.PackageManagement.PowerShellCmdlets.csproj:
```diff
@@ -12,11 +12,9 @@
     <FileAlignment>512</FileAlignment>
-    <NuGetPackageImportStamp>44966557</NuGetPackageImportStamp>
-    <PackagesDirectory>$(UserProfile)\.nuget\packages</PackagesDirectory>
+    <Description>NuGet's PowerShell Cmdlets for the Visual Studio client.</Description>
```
```suggestion
    <Description>NuGet's PowerShell cmdlets for the Visual Studio client.</Description>
```

dominoFire reviewed (approved):
Looks good. Some minor comments left.

nkolev92 commented:
Thanks @dominoFire Applied your suggestions. 

@donnie-msft Can please either take another look or dismiss the stale review.

donnie-msft commented on test/NuGet.Core.FuncTests/Dotnet.Integration.Test/Dotnet.Integration.Test.csproj:
```diff
@@ -5,6 +5,7 @@
   <PropertyGroup>
     <TargetFrameworks>$(NETCoreTargetFramework)</TargetFrameworks>
     <TestProject>true</TestProject>
+    <Description>Integration tests for NuGet-powered dotnet CLI comands such as pack/restore/list package and dotnet nuget.</Description>
```
typo:commands

nkolev92 replied:
fixed.

donnie-msft reviewed (approved):
Just one more little typo!

nkolev92 committed:
add descriptions for every project
